### PR TITLE
Update bindings

### PR DESCRIPTION
**Pull request by hwsmm**
I actually didn't mean to work on this. I had updated my system SDL package to master, and osu! was straight up broken, so I tried updating bindings to spend some free time.

I don't mean to take over maintaining bindings. This is probably just a one-time thing unless my accidental package update breaks things again...

There may be some things I may have missed since this is my first time updating bindings. Please let me know if there are any!

### CODE REVIEW

**Review by smoogipoo — APPROVED** [review]

**Comment by smoogipoo** [comment]
> and osu! was straight up broken

This shouldn't happen because we package our own SDL3 lib. Would be curious if you could run osu! with `LD_DEBUG=libs` and paste that output, as it's very concerning.

**Comment by hwsmm** [comment]
I guess I should have been more clear. I use my system SDL3 *intentionally* because SDL3 we build is currently missing PipeWire support and some minor things I want.

It uses bundled SDL3 without my script, so you don't need to worry about that.

File: SDL3-CS/SDL3/ClangSharp/SDL_mouse.g.cs [hunk 1]
```diff
@@ -78,15 +78,15 @@ public static unsafe partial class SDL3
...
         public static extern uint SDL_GetMouseState(float* x, float* y);
 
         [DllImport("SDL3", CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
-        [return: NativeTypeName("Uint32")]
+        [return: NativeTypeName("SDL_MouseButtonFlags")]
         public static extern uint SDL_GetGlobalMouseState(float* x, float* y);
 
         [DllImport("SDL3", CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
-        [return: NativeTypeName("Uint32")]
+        [return: NativeTypeName("SDL_MouseButtonFlags")]
```
Comment (Susko3) [inline SDL3-CS/SDL3/ClangSharp/SDL_mouse.g.cs:89]: `SDL_MouseButtonFlags` should be added as a `[Typedef] enum`, replacing the manually-added `SDLButtonMask`.

You can search for `[Typedef]` to see how to add it. After adding it, just rerun the python script.

File: SDL3-CS/SDL3/ClangSharp/SDL_pen.g.cs [hunk 2]
```diff
@@ -85,7 +85,7 @@ public static unsafe partial class SDL3
         public static extern byte* Unsafe_SDL_GetPenName(SDL_PenID instance_id);
 
         [DllImport("SDL3", CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
-        [return: NativeTypeName("Uint32")]
+        [return: NativeTypeName("SDL_PenCapabilityFlags")]
         public static extern uint SDL_GetPenCapabilities(SDL_PenID instance_id, SDL_PenCapabilityInfo* capabilities);
```
Comment (Susko3) [inline SDL3-CS/SDL3/ClangSharp/SDL_pen.g.cs:89]: `SDL_PenCapabilityFlags` should also be added as a `[Typedef] enum`.

File: SDL3-CS/SDL3/ClangSharp/SDL_video.g.cs [hunk 3]
```diff
@@ -236,6 +240,9 @@ public static unsafe partial class SDL3
         [return: NativeTypeName("Uint32")]
         public static extern uint SDL_GetWindowPixelFormat(SDL_Window* window);
 
+        [DllImport("SDL3", CallingConvention = CallingConvention.Cdecl, ExactSpelling = true)]
+        public static extern SDL_Window** SDL_GetWindows(int* count);
```
Comment (Susko3) [inline SDL3-CS/SDL3/ClangSharp/SDL_video.g.cs:244]: This fits the `SDLPointerArray` pattern.

**Review by Susko3 — CHANGES_REQUESTED** [review]
Missing some SDL3-CS specifics, seems fine otherwise. I can do the changes if you wish.

**Comment by hwsmm** [comment]
I applied requested changes and updated bindings!

**Review by Susko3 — APPROVED** [review]
LGTM, checked that generating the bindings locally matches changes in this PR.

**Pull request merged**